akka-distributed-data/src/main/scala/akka/cluster/ddata/GCounter.scala

```diff
@@ -77,14 +79,19 @@ final class GCounter private[akka] (
   /**
    * INTERNAL API
    */
-  private[akka] def increment(key: UniqueAddress, delta: BigInt): GCounter = {
-    require(delta >= 0, "Can't decrement a GCounter")
-    if (delta == 0) this
-    else state.get(key) match {
-      case Some(v) ⇒
-        val tot = v + delta
-        assignAncestor(new GCounter(state + (key → tot)))
-      case None ⇒ assignAncestor(new GCounter(state + (key → delta)))
+  private[akka] def increment(key: UniqueAddress, n: BigInt): GCounter = {
+    require(n >= 0, "Can't decrement a GCounter")
+    if (n == 0) this
+    else {
+      val nextValue = state.get(key) match {
+        case Some(v) ⇒ v + n
+        case None    ⇒ n
+      }
+      val newDelta = delta match {
+        case Some(d) ⇒ Some(new GCounter(d.state + (key → nextValue)))
+        case None    ⇒ Some(new GCounter(Map(key → nextValue)))
+      }
+      assignAncestor(new GCounter(state + (key → nextValue), newDelta))
```

Comment:
Hm I see, goes to show I don't know enough about these parts to properly review :~